### PR TITLE
[Core] Minor clean up and additional info on ResidualCriteria

### PR DESCRIPTION
**Pull request by loumalouomega**
The info is displayed in case of EchoLevel >= 2. I did this clean up looking for the origin of the problem on #5052

No change on behavior

### CODE REVIEW

File: kratos/solving_strategies/convergencecriterias/residual_criteria.h [hunk 1]
```diff
@@ -177,6 +175,7 @@ class ResidualCriteria
             const TDataType float_size_residual = static_cast<TDataType>(size_residual);
             const TDataType absolute_norm = (mCurrentResidualNorm/float_size_residual);
 
+            KRATOS_INFO_IF("RESIDUAL CRITERION", this->GetEchoLevel() > 1 && rModelPart.GetCommunicator().MyPID() == 0) << " :: [ Initial residual norm = " << mInitialResidualNorm << "; Current residual norm =  " << mCurrentResidualNorm << "]" << std::endl;
```
Comment (jcotela) [inline kratos/solving_strategies/convergencecriterias/residual_criteria.h:178]: Note that the MyPID() check is not strictly needed, KRATOS_INFO_IF only prints on rank 0 now for MPI

**Review by jcotela — APPROVED** [review]
:+1:

File: kratos/solving_strategies/convergencecriterias/residual_criteria.h (hunk 1 above)
Comment (philbucher) [reply kratos/solving_strategies/convergencecriterias/residual_criteria.h:178]: I wanted to write the same but I think it makes more sense to update the loggers all at once
=> removing the rank-check and passing the DataComm

**Review by philbucher — APPROVED** [review]
sorry for the late response

**Comment by loumalouomega** [comment]
Thanks

**Pull request merged**